fletcher/_algorithms.py

```diff
@@ -246,3 +246,65 @@ def all_op(arr, skipna):
         return _all_op_nonnull(len(arr), arr.buffers()[1])
     # skipna is not relevant in the Pandas behaviour
     return _all_op(len(arr), *arr.buffers())
+
+
+def max_fletcher_array(fr_arr):
+    if len(fr_arr) == 0:
+        return None
+
+    def max_of_one_chunk(chunk):
+        arr_buff = np.frombuffer(
+            chunk.buffers()[1],
+            dtype=chunk.type.to_pandas_dtype(),
+            count=len(chunk),
+            offset=chunk.offset,
+        )
+        return max_ma(arr_buff, chunk.buffers()[0])
+
+    return max([max_of_one_chunk(ch) for ch in fr_arr.data.chunks])
```

Comment:
`max` works on generator - no need for list
```suggestion
    return max(max_of_one_chunk(ch) for ch in fr_arr.data.chunks)
```